include/RevCore.h

```diff
@@ -671,7 +671,10 @@ class RevCore {
 
   EcallStatus ECALL_dump_thread_mem();         // 9006, dump_thread_mem()
   EcallStatus ECALL_dump_thread_mem_to_file(); // 9007, dump_thread_mem_to_file(const char* outputFile)
-  // clang-format on
+
+  // =============== REV print utilities
+  EcallStatus ECALL_fast_printf();             // 9010, rev_fast_printf(const char *, uint64_t a1=0, uint64_t a2=0, uint64_t a3=0, uint64_t a4=0, uint64_t a5=0, uint64_t a6=0)
+    // clang-format on
```

Comment:
Variadic templates would make it simple to pass an arbitrary number of arguments. `std::forward` is used to pass the rvalue-ness of the arguments.

Reply:
Well, since you're accessing each `ECALL` argument (`a0-a7`) separately, the variadics may be only useful at the C++ level but not lower-level.

You may also have to be concerned with `XLEN` being 32- or 64-bits, although 64-bits (`uint64_t`) should be safe at the C++ interface level.

Reply:
By the way, even if `XLEN==32`, if you use `GetX<uint64_t>(reg)`, the 32-bit register will be zero-extended to 64 bits. Similarly. `GetX<int64_t>(reg)` will sign-extend the 32-bit register to 64 bits.

Using size-specific types like `uint64_t` is strongly encouraged instead of using types like `unsigned` or `long`.

Reply:
> I originally was using a variadic but they did not work on that version of REV ( much earlier this year ). I will try it again on the latest version of REV. I would be good to confirm that the variadics now work.

It should not depend on the Rev version. Variadic templates are a C++11 feature and we've been compiling Rev with C++17 for over a year. If you use non-type template arguments like `<uint64_t... args>` then a variable number of integer arguments should be passed to the function.